pkg/provisioner/ironic/testserver/server.go

```diff
@@ -81,6 +82,17 @@ func (m *MockServer) Response(pattern string, payload string) *MockServer {
 	return m
 }
 
+// ResponseJSON marshals the JSON object as payload returned by the response
+// handler
+func (m *MockServer) ResponseJSON(pattern string, payload interface{}) *MockServer {
+	content, err := json.Marshal(payload)
+	if err != nil {
+		panic(err)
```

Comment:
The server has a handle to the `testing.T`, so we could just report a fatal error to stop the current test, instead of panicking.

Reply:
Yes, sounds better